src/standalone/Soil/Biogeochemistry/Biogeochemistry.jl

```diff
@@ -467,12 +485,13 @@ function ClimaLSM.boundary_flux(
     Δz::ClimaCore.Fields.Field,
     Y::ClimaCore.Fields.FieldVector,
     p::NamedTuple,
-    t::FT,
-)::ClimaCore.Fields.Field where {FT}
+    t,
+)::ClimaCore.Fields.Field
+    FT = eltype(Δz)
```

Comment:
I think my pref is to get FT from the BC type. but we can discuss! otherwise Im not sure the BC's need to have a parametric type at all.

we could also make clear in the BC docs (future PR) that FT is used to convert functions of time into the right type, or something.